### ISSUE TITLE
Will the PhotoShop script for generating the fake images be released?

### ISSUE THREAD

**Issue by zjxgithub** [PeterWang512/FALdetector#3]
Hi, thanks for sharing the model! 
I wonder if you will also release the PhotoShop script for generating the fake images? Thanks!

**Comment by jalbertbowden** [comment]
I came to look @ the photoshop script too! Now I'm thinking that the scripts are the python files in the repository, and the title is just clever.

**Comment by PeterWang512** [comment]
Yes, we are planning to release the PhotoShop script to automatically generate FAL-warped faces

**Comment by zhangqizky** [comment]
> Yes, we are planning to release the PhotoShop script to automatically generate FAL-warped faces

It's awesome!!!

**Comment by jalbertbowden** [comment]
Can't wait! Very interested to see what y'all did; I've done some `.jsx` scripting in the past, but nothing very high level/beyond using action panes/panels.

**Comment by jaavant** [comment]
Been almost a year since the last post, is there any plan to still release the script?

**Comment by PeterWang512** [comment]
Sorry for the delay. Below is a script for automating warps for an image, and one can save it as a .jsx file to run it on Photoshop.
To begin with, Photoshop allows users to automate many processes with javascript/applescript/vbscript, and here is the link to guides: https://www.adobe.com/devnet/photoshop/scripting.html.
(Note that the guide above not necessarily covers all operations that can be automated; for example, filters like face-aware liquify.)

As a quick start, you can open up an image with face in photoshop, click on File->Scripts->Browse, and then select the .jsx file to run it.

Hope this helps!

`var param_leftEyeSize = Math.random()*2 - 1;
var param_rightEyeSize = Math.random()*2 - 1;
var param_leftEyeHeight = Math.random()*2 - 1;
var param_rightEyeHeight = Math.random()*2 - 1;
var param_leftEyeWidth = Math.random()*2 - 1;
var param_rightEyeWidth = Math.random()*2 - 1;
var param_leftEyeTilt = Math.random()*2 - 1;
var param_rightEyeTilt = Math.random()*2 - 1;
var param_eyeDistance = Math.random()*2 - 1;
var param_smile = Math.random()*2 - 1;
var param_upperLip = Math.random()*2 - 1;
var param_lowerLip = Math.random()*2 - 1;
var param_mouthWidth = Math.random()*2 - 1;
var param_noseWidth = Math.random()*2 - 1;
var param_faceWidth = Math.random()*2 - 1;
var param_chinHeight = Math.random()*2 - 1;
var param_jawShape = Math.random()*2 - 1;
var param_mouthHeight = Math.random()*2 - 1;
var param_noseHeight = Math.random()*2 - 1;
var param_foreheadHeight = Math.random()*2 - 1;

var idLqFy = charIDToTypeID( "LqFy" );
    var desc16 = new ActionDescriptor();
    var idLqMe = charIDToTypeID( "LqMe" );
    var idfaceMesh = stringIDToTypeID( "faceMesh" );
        var desc17 = new ActionDescriptor();
        var idfaceDescriptorVersion = stringIDToTypeID( "faceDescriptorVersion" );
        desc17.putInteger( idfaceDescriptorVersion, 2 );
        var idfaceMeshVersion = stringIDToTypeID( "faceMeshVersion" );
        desc17.putInteger( idfaceMeshVersion, 2 );
        var idfaceInfoList = stringIDToTypeID( "faceInfoList" );
            var list1 = new ActionList();
                var desc18 = new ActionDescriptor();

var idfeatureValues = stringIDToTypeID( "featureValues" );
                    var desc20 = new ActionDescriptor();
                    var idleftEyeSize = stringIDToTypeID( "leftEyeSize" );
                    desc20.putDouble( idleftEyeSize, param_leftEyeSize );
                    var idrightEyeSize = stringIDToTypeID( "rightEyeSize" );
                    desc20.putDouble( idrightEyeSize, param_rightEyeSize );
                    var idleftEyeHeight = stringIDToTypeID( "leftEyeHeight" );
                    desc20.putDouble( idleftEyeHeight, param_leftEyeHeight );
                    var idrightEyeHeight = stringIDToTypeID( "rightEyeHeight" );
                    desc20.putDouble( idrightEyeHeight, param_rightEyeHeight );
                    var idleftEyeWidth = stringIDToTypeID( "leftEyeWidth" );
                    desc20.putDouble( idleftEyeWidth, param_leftEyeWidth );
                    var idrightEyeWidth = stringIDToTypeID( "rightEyeWidth" );
                    desc20.putDouble( idrightEyeWidth, param_rightEyeWidth );
                    var idleftEyeTilt = stringIDToTypeID( "leftEyeTilt" );
                    desc20.putDouble( idleftEyeTilt, param_leftEyeTilt );
                    var idrightEyeTilt = stringIDToTypeID( "rightEyeTilt" );
                    desc20.putDouble( idrightEyeTilt, param_rightEyeTilt );
                    var ideyeDistance = stringIDToTypeID( "eyeDistance" );
                    desc20.putDouble( ideyeDistance, param_eyeDistance );
                    var idsmile = stringIDToTypeID( "smile" );
                    desc20.putDouble( idsmile, param_smile );
                    var idupperLip = stringIDToTypeID( "upperLip" );
                    desc20.putDouble( idupperLip, param_upperLip );
                    var idlowerLip = stringIDToTypeID( "lowerLip" );
                    desc20.putDouble( idlowerLip, param_lowerLip );
                    var idmouthWidth = stringIDToTypeID( "mouthWidth" );
                    desc20.putDouble( idmouthWidth, param_mouthWidth );
                    var idnoseWidth = stringIDToTypeID( "noseWidth" );
                    desc20.putDouble( idnoseWidth, param_noseWidth );
                    var idfaceWidth = stringIDToTypeID( "faceWidth" );
                    desc20.putDouble( idfaceWidth, param_faceWidth );
                    var idchinHeight = stringIDToTypeID( "chinHeight" );
                    desc20.putDouble( idchinHeight, param_chinHeight );
                    var idjawShape = stringIDToTypeID( "jawShape" );
                    desc20.putDouble( idjawShape, param_jawShape );
                    var idmouthHeight = stringIDToTypeID( "mouthHeight" );
                    desc20.putDouble( idmouthHeight, param_mouthHeight );
                    var idnoseHeight = stringIDToTypeID( "noseHeight" );
                    desc20.putDouble( idnoseHeight, param_noseHeight );
                    var idforeheadHeight = stringIDToTypeID( "foreheadHeight" );
                    desc20.putDouble( idforeheadHeight, param_foreheadHeight );
                desc18.putObject( idfeatureValues, idfeatureValues, desc20 );
                var idfeatureDisplacements = stringIDToTypeID( "featureDisplacements" );
                    var desc21 = new ActionDescriptor();
                var idfeatureDisplacements = stringIDToTypeID( "featureDisplacements" );
                desc18.putObject( idfeatureDisplacements, idfeatureDisplacements, desc21 );
            var idfaceInfo = stringIDToTypeID( "faceInfo" );
            list1.putObject( idfaceInfo, desc18 );
        desc17.putList( idfaceInfoList, list1 );
    var idfaceMesh = stringIDToTypeID( "faceMesh" );
    desc16.putObject( idfaceMesh, idfaceMesh, desc17 );
executeAction( idLqFy, desc16, DialogModes.NO );`
